glue/utils/qt/autocomplete_widget.py

```diff
@@ -0,0 +1,127 @@
+# Code adapted from:
+#
+# http://rowinggolfer.blogspot.de/2010/08/qtextedit-with-autocompletion-using.html
+#
+# and based on:
+#
+# http://qt-project.org/doc/qt-4.8/tools-customcompleter.html
+
+from ...external.qt import QtGui, QtCore
+
+
+__all__ = ["CompletionTextEdit"]
+
+try:  # Python 2
+
+    QString = QtCore.QString
```

Comment:
Ah I was using this as standalone for testing so maybe the API wasn't getting set. I'll see if I can simplify all this now it's being used in glue.